### PR TITLE
(Doc+) SIEM + frozen tier compatibility 

### PR DESCRIPTION
**Pull request by stefnestor**
👋🏽 howdy, team! 

1. Since [ES Data Tiers](https://www.elastic.co/guide/en/elasticsearch/reference/current/data-tiers.html) are sequentially hardware performant, because we call out `cold` this appends the same rules apply to `frozen` (related to `securitySolution:excludeColdAndFrozenTiersInAnalyzer` from [kibana#172162](https://github.com/elastic/kibana/pull/172162)).  Kindly see examples 
    - [kibana#139969](https://github.com/elastic/kibana/issues/139969) for linked internal examples of e.g. alert indices `.alerts-security*` being hosted in frozen breaking SIEM. 
    - [kb#39606517](https://support.elastic.co/knowledge/39606517) where frozen indices containing future dates catch in the Rule lookback window and therefore can degrade product performance. 
2. It looks like there's a duplicate section in the [SIEM](https://www.elastic.co/guide/en/security/master/detection-engine-overview.html#cold-tier-detections) and [Kibana](https://www.elastic.co/guide/en/kibana/current/xpack-siem.html#_compatibility_with_cold_tier_nodes) docs from [security-docs#487](https://github.com/elastic/security-docs/pull/487) and [kibana#103151](https://github.com/elastic/kibana/pull/103151), respectively, where the latter had additional phrasing so cross-pollinated. 
    - ( Paper-trail FYI for my team: [security-docs#551](https://github.com/elastic/security-docs/issues/551) later appears to confirm indicator rules don't support cold, to confirm between PR's which is latest state in (2). It does also suggest indicator match rules do not support cross cluster search, but that was purposely later removed in [security-docs#3054](https://github.com/elastic/security-docs/pull/3054) for v8.7 . )

Preview:
- [Detections and alerts](https://security-docs_bk_5564.docs-preview.app.elstc.co/guide/en/security/master/detection-engine-overview.html)

### CODE REVIEW

File: docs/detections/detection-engine-intro.asciidoc
Comment (nastasha-solomon) [inline docs/detections/detection-engine-intro.asciidoc:77]: Adds a bit more consistency in the way we talk about rules on this page.

```suggestion
* Index patterns specified in the definition of indicator match rules 
```

**Review by nastasha-solomon — COMMENTED** [review]
Thanks for making these changes, @stefnestor! Couple of questions, in addition to the comments that I left:
- Do these changes need to be reflected in the [Serverless version](https://www.elastic.co/docs/current/serverless/security/detection-engine-overview) of this page too? If so, I'm happy to add those updates to this PR. 
- How far back should we port these changes? For example, is this information relevant to `8.7` - `8.15`? 

Thanks again for your help with keeping the Security docs current and correct!

File: docs/detections/detection-engine-intro.asciidoc
Comment (stefnestor) [reply docs/detections/detection-engine-intro.asciidoc:68]: Agreed not sure we need the full caveat but FWIW, the ES docs draw the distinction:

> [Data Tiers](https://www.elastic.co/guide/en/elasticsearch/reference/current/data-tiers.html) 
> image

**Comment by stefnestor** [comment]
Okay, read again! 🙏

File: docs/detections/detection-engine-intro.asciidoc
Comment (nastasha-solomon) [inline docs/detections/detection-engine-intro.asciidoc:68]: Oops, forgot to fix this earlier.

```suggestion
Cold and frozen {ref}/data-tiers.html[data tiers] hold time series data that is only accessed occasionally. In {stack} version >=7.11.0, {elastic-sec} supports cold tier data for the following {es} indices:
```

File: docs/detections/detection-engine-intro.asciidoc
Comment (rylnd) [inline docs/detections/detection-engine-intro.asciidoc:68]: I agree with adding mention of frozen tiers for clarity, but it doesn't seem like this line things much clearer. If I'm understanding correctly, we say:

1. X supports cold tier
2. Y and Z do not support cold or frozen tier

It seems like we should be explicit on whether X (which is the list on lines 70-72) supports frozen tier or not.

Comment (stefnestor) [reply docs/detections/detection-engine-intro.asciidoc:68]:
```suggestion
Cold and frozen {ref}/data-tiers.html[data tiers] hold time series data that is only accessed occasionally. In {stack} version >=7.11.0, {elastic-sec} supports cold but not frozen tier data for the following {es} indices:
```

Does this look good?

Comment (rylnd) [reply docs/detections/detection-engine-intro.asciidoc:68]: It addresses my issue,  but I defer to @marshallmain to confirm that this statement is correct

Comment (marshallmain) [reply docs/detections/detection-engine-intro.asciidoc:68]: This is roughly correct but I think we on the dev/product side need to come up with a more precise concept of what exactly we support because separating things by index pattern isn't sufficient granularity IMO. We do support frozen tier data existing in index patterns queried by detection rules as long as the frozen data is not in the time range queried by the rule. In general for the use cases listed here it's important that frozen tier data isn't queried regularly, i.e. that the time range on the query excludes frozen tier data (or a _tier filter? I can't remember if we tested to confirm that optimization works or not), but cold/frozen tier data is allowed to live in those index patterns.

So for detection rules, we might say "frozen tier data must not be in the time range queried by the rule or it must be explicitly excluded with a _tier filter". For new terms rules, we should also specify that frozen tier data can't be in the (interval + lookback) OR the history window. Similarly for the "data sources selector on various pages", the frozen data only needs to be excluded from the query.

Comment (marshallmain) [reply docs/detections/detection-engine-intro.asciidoc:68]: I don't think we need to hold up this PR to rework the concepts here though.

**Review by nastasha-solomon — APPROVED** [review]

File: docs/detections/detection-engine-intro.asciidoc
Comment (nastasha-solomon) [reply docs/detections/detection-engine-intro.asciidoc]: @marshallmain I'm tracking your feedback in a private docs issue so we don't forget to revisit these docs in the future.

https://github.com/elastic/security-docs-internal/issues/27

**Pull request merged**
